lib/utils.js

```
@@ -496,10 +496,24 @@ function isTvOS (platformName) {
   return _.toLower(platformName) === _.toLower(PLATFORM_NAME_TVOS);
 }
 
+/**
+ * Returns true if the urlString is localhost
+ * @param {?string} urlString
+ * @returns {boolean} Return true if the urlString is localhost
+ */
+function isLocalHost (urlString) {
+  try {
+    const {hostname} = new URL(urlString);
+    return hostname.includes('localhost')
+      || ['127.0.0.1', '[::1]', '[::ffff:7f00:1]'].includes(hostname);
+  } catch {}
```

why should we ignore the error?

---

to make it safe we could

```
if (!_.isString(address)) {
  return false;
}
```

---

We could write a warning message here